index.php

```diff
@@ -59,24 +55,42 @@ function processSlug() {
     function createLink() {
         global $base_url, $random_chars, $slug_length, $db, $pw_create, $accepts_json;
 
-        $url = ltrim($_SERVER['REQUEST_URI'], '/');
+        $url = isset($_POST['url']) ? '' : $_GET['url'];
+        $custom_slug = isset($_POST['slug']) ? '' : $_GET['slug'];
```

Comment:
`$custom_slug` should be escaped before it's used in `mysql_query` on lines 84 and 97.

```suggestion
        $custom_slug = mysqli_real_escape_string($db, isset($_POST['slug']) ? '' : $_GET['slug']);
```